client/components/foldable-card/style.scss

```diff
@@ -135,6 +136,20 @@ button.foldable-card__action {
 	display: none;
 }
 
+.foldable-card.is-smooth .foldable-card__content {
+	display: block;
+	overflow: hidden;
+
+	transition: max-height 0.25s cubic-bezier(0.175, 0.885, 0.32, 1.275), padding 0.2s ease-in;
+	max-height: 0;
+}
+
+.foldable-card.is-expanded.is-smooth .foldable-card__content {
+	height: auto;
+	max-height: 250px; // fallback if not set inline with js
+	transition: max-height 0.25s cubic-bezier(0.175, 0.885, 0.32, 1.275), padding 0.2s ease-in;
+}
+
 .foldable-card.is-expanded .foldable-card__content {
 	display: block;
 	padding: 16px;
```

Comment:
In a "max-height" transitioning container, there should **not** be any padding (or margin). The content inside needs to be responsible for any padding or margins, otherwise when the transitioning content container first appears, it immediately has padding (causing this weird padding effect, where in this case we're transitioning the padding in (which we don't want to do), causing this weird padding-sliding-in effect, shown below). 

![Screen Capture on 2022-09-16 at 12-22-56](https://user-images.githubusercontent.com/11078128/190685787-6517fd76-93c4-49c4-88b8-e93ea57e60f7.gif)

So therefore, as a fix, we should only apply padding to the `foldable-card__content` container if it does **not** have the `.is-smooth` class, for example, like this:

```
.foldable-card.is-expanded .foldable-card__content {
	display: block;
	// padding: 16px; // remove this.
	border-top: 1px solid var(--color-neutral-5);
}

.foldable-card.is-expanded:not(.is-smooth) .foldable-card__content {
	padding: 16px;
}

.foldable-card.is-compact .foldable-card.is-expanded:not(.is-smooth) .foldable-card__content {
	padding: 8px;
}
```

And then for the case of the devdocs example (or for any case where someone is going to utilize the smooth prop), we/they would then be responsible for adding the padding to their _contents inside_ `.foldable-card__content`, but not to `.foldable-card__content itself. For example:
```
<div style="padding: 16px">
    <p>These are its contents</p>
    <p>And some more</p>
</div>
```
Does that make sense?, it's kinda tricky to explain, so if it is still unclear, we can pair up or something and I can demonstrate further if you want. 😉

Reply:
Thank you! I totally agree and thanks for a great explanation!